mlflow/models/docker_utils.py

```diff
@@ -13,27 +13,26 @@
 
 _DOCKERFILE_TEMPLATE = """
 # Build an image that can serve mlflow models.
-FROM ubuntu:16.04
+FROM {base_image}
 
 RUN apt-get -y update && apt-get install -y --no-install-recommends \
          wget \
          curl \
          nginx \
          ca-certificates \
          bzip2 \
-         build-essential \
-         cmake \
-         openjdk-8-jdk \
          git-core \
-         maven \
-    && rm -rf /var/lib/apt/lists/*
 
 # Download and setup miniconda
 RUN curl https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh >> miniconda.sh
 RUN bash ./miniconda.sh -b -p /miniconda; rm ./miniconda.sh;
 ENV PATH="/miniconda/bin:$PATH"
-ENV JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64
 ENV GUNICORN_CMD_ARGS="--timeout 60 -k gevent"
+
+{java_libraries}
```

Comment:
rename as `java_setup` since it sets up env arguments, downloads dependencies ... etc